### PR TITLE
Add more flexibility to generate_policies

### PR DESCRIPTION
**Pull request by MichaelEizaguirre**
These changes allow for generate policies to have more customization, instead of passing in as a flag how many From, When, To rules you want to generate in a rule, these changes allow the user to choose between numPaths, numSourceIP, numNamespaces, and numValues to create those rules in a policy. These changes will also allow more flexibility when in the future we need to add more options for rules like requestPrinciple and so fourth

### CODE REVIEW

**Comment by MichaelEizaguirre** [comment]
@yangminzhu @carolynhu

**Comment by MichaelEizaguirre** [comment]
I did look into it more, it does not seem like there would be a good way to tell runner.py to print out the security labels without passing in a security flag like you mention. It could be a simple boolean flag security flag. The main question is since we are calling generate_policies before running runner.py we need to store the label in some temp location so that when we call runner.py with the security flag runner.py can fetch the data and create the label. Do you have any idea of where we could store the generate_policies label data?

**Comment by carolynhu** [comment]
> I did look into it more, it does not seem like there would be a good way to tell runner.py to print out the security labels without passing in a security flag like you mention. It could be a simple boolean flag security flag. The main question is since we are calling generate_policies before running runner.py we need to store the label in some temp location so that when we call runner.py with the security flag runner.py can fetch the data and create the label. Do you have any idea of where we could store the generate_policies label data?

Yeah, that's why it's hard to construct those labels without the main login in runner.py. For the current test flow, you can save it as an Environment Variable, that would be the easiest way I could think of.

File: perf/benchmark/security/generate_policies/generate.go [hunk 1]
```diff
@@ -21,22 +21,25 @@ import (
...
 
-func (operationGenerator) generate(kind string, num int, _ string) *authzpb.Rule {
+func (operationGenerator) generate(_ string, ruleMap map[string]int) *authzpb.Rule {
 	rule := &authzpb.Rule{}
 	var listOperation []*authzpb.Rule_To
 
-	for i := 0; i < num; i++ {
-		path := fmt.Sprintf("/invalid-path-%d", i)
+	numPaths := ruleMap["numPaths"]
+	if numPaths > 0 {
```
[the remark this answers is not in the document]
Comment (MichaelEizaguirre) [reply perf/benchmark/security/generate_policies/generate.go:35]: Good point

File: perf/benchmark/security/generate_policies/generate_policies.go [hunk 2]
```diff
@@ -159,54 +151,106 @@ func createPolicyHeader(namespace string, name string, kind string) *MyPolicy {
...
-	policyType := flag.String("policyType", "AuthorizationPolicy", "The type of security policy. Supported value: AuthorizationPolicy")
-	actionPtr := flag.String("action", "DENY", "Type of action. Supported values: DENY, ALLOW")
-	numPoliciesPtr := flag.Int("numPolicies", 1, "Number of policies wanted")
-
-	ruleToOccurancesPtr := make(map[string]*int)
-	ruleToOccurancesPtr["when"] = flag.Int("when", 1, "Number of when condition wanted")
-	ruleToOccurancesPtr["to"] = flag.Int("to", 1, "Number of To operations wanted")
-	ruleToOccurancesPtr["from"] = flag.Int("from", 1, "Number of From sources wanted")
+	securityPtr := flag.String("generate_policy", "numPolicies:1", `List of key value pairs separated by commas.
+	Supported options: namespace:\"string\", action:DENY/ALLOW, policyType:AuthorizationPolicy,
```
[the remark this answers is not in the document]
Comment (MichaelEizaguirre) [reply perf/benchmark/security/generate_policies/generate_policies.go:219]: No need for that I will remove it

**Pull request merged**